engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/EndEventTransformer.java

```diff
@@ -31,11 +32,14 @@ public void transform(final EndEvent element, final TransformContext context) {
     final var currentProcess = context.getCurrentProcess();
     final var endEvent = currentProcess.getElementById(element.getId(), ExecutableEndEvent.class);
 
+    endEvent.setEventType(BpmnEventType.NONE);
+
     if (!element.getEventDefinitions().isEmpty()) {
       transformEventDefinition(element, context, endEvent);
     }
 
     if (isMessageEvent(element) && hasTaskDefinition(element)) {
+      endEvent.setEventType(BpmnEventType.MESSAGE);
       jobWorkerElementTransformer.transform(element, context);
     }
```

Comment:
🔧 In the future we'd like to support message end events that do not require a task definition. So in order to keep this future-proof we should put the `setEventType` in a separate if statement from this as it could be a message event without task definition.

Reply:
I've changed both `EndEventTransformer` and `IntermediateThrowEventTransformer`.